src/core/coap/coap.cpp

```diff
@@ -139,6 +139,64 @@ Message *CoapBase::NewMessage(const Message::Settings &aSettings)
     return message;
 }
 
+Message *CoapBase::NewPriorityConfirmablePostMessage(const char *aUriPath)
```

Comment:
Is it possible to create a `Tmf::MessageBuilder` class instead of adding all these helpers to `CoapBase`?

Reply:
These method can be used with `TmfAgent` but also in some cases with `CoapSecure` (and also with `BakcboneTfmAgent` which is its own type and not related to `TmfAgent`) so that's why I added them in  `CoapBase` to be usable under all different sub-classes.

The goal is this PR is to refactor common patterns used in many places into helper methods (to avoid repeating code and reduce code-size). Adding a `Builder` may be something worth considering in future PRs, though I think to some extend the `Coap::Message` itself provides  similar model (and I am not sure if it can help reduce code size).

Reply:
Agree.